x-pack/plugins/observability_solution/apm/public/components/app/service_inventory/multi_signal_inventory/index.tsx

```diff
@@ -74,9 +78,72 @@ function useServicesEntitiesMainStatisticsFetcher() {
   return { mainStatisticsData: data, mainStatisticsStatus: status };
 }
 
-export const MultiSignalInventory = () => {
+function useServicesEntitiesDetailedStatisticsFetcher({
+  mainStatisticsFetch,
+  services,
+}: {
+  mainStatisticsFetch: ReturnType<typeof useServicesEntitiesMainStatisticsFetcher>;
+  services: ServiceListItem[];
+}) {
+  const {
+    query: { rangeFrom, rangeTo, environment, kuery },
+  } = useApmParams('/services');
+
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const dataSourceOptions = usePreferredDataSourceAndBucketSize({
+    start,
+    end,
+    kuery,
+    type: ApmDocumentType.ServiceTransactionMetric,
+    numBuckets: 20,
+  });
+
+  const { mainStatisticsData, mainStatisticsStatus } = mainStatisticsFetch;
+
+  const timeseriesDataFetch = useProgressiveFetcher(
+    (callApmApi) => {
+      const serviceNames = services.map(({ serviceName }) => serviceName);
+
+      if (
+        start &&
+        end &&
+        serviceNames.length > 0 &&
+        mainStatisticsStatus === FETCH_STATUS.SUCCESS &&
+        dataSourceOptions
+      ) {
+        return callApmApi('POST /internal/apm/entities/services/detailed_statistics', {
+          params: {
+            query: {
+              environment,
+              kuery,
+              start,
+              end,
+              documentType: dataSourceOptions.source.documentType,
+              rollupInterval: dataSourceOptions.source.rollupInterval,
+              bucketSizeInSeconds: dataSourceOptions.bucketSizeInSeconds,
+            },
+            body: {
+              // Service name is sorted to guarantee the same order every time this API is called so the result can be cached.
+              serviceNames: JSON.stringify(serviceNames.sort()),
+            },
+          },
+        });
+      }
+    },
+    // only fetches detailed statistics when requestId is invalidated by main statistics api call or offset is changed
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [mainStatisticsData.requestId, services],
```

Comment:
What is the dependency that causes the API to be unnecessarily called?

Reply:
This is similar to the request we have in the services overview (`POST /internal/apm/services/detailed_statistics`)  It's `start`, `end`, `environment`, `kuery`, and `dataSourceOptions` in this case